src/main/java/frc/robot/subsystems/swerve/SwerveSubsystem.java

```diff
@@ -178,7 +206,83 @@ public void periodic() {
     }
+    Logger.recordOutput("Swerve Pose", pose);
+
+    visionIO.updateInputs(visionInputs, new Pose3d(pose));
+    Logger.processInputs("Vision", visionInputs);
+    PhotonPipelineResult result =
+        new PhotonPipelineResult(visionInputs.latency, visionInputs.targets);
+    result.setTimestampSeconds(visionInputs.timestamp);
+    try {
+      var estPose =
+          VisionHelper.update(
+                  result,
+                  tagFieldLayout,
+                  PoseStrategy.MULTI_TAG_PNP_ON_COPROCESSOR,
+                  PoseStrategy.LOWEST_AMBIGUITY)
+              .get();
+      var visionPose =
+          VisionHelper.update(
+                  result,
+                  tagFieldLayout,
+                  PoseStrategy.MULTI_TAG_PNP_ON_COPROCESSOR,
+                  PoseStrategy.LOWEST_AMBIGUITY)
+              .get()
+              .estimatedPose;
+      Logger.recordOutput("Vision Pose", visionPose);
+      estimator.addVisionMeasurement(
+          visionPose.toPose2d(),
+          visionInputs.timestamp,
+          VisionHelper.findVisionMeasurements(estPose));
+    } catch (NoSuchElementException e) {
+    }
+
+    // From 6995 - untested
+    // try {
+    //   var estPose =
+    //       VisionHelper.update(
+    //               result, tagFieldLayout, PoseStrategy.MULTI_TAG_PNP_ON_COPROCESSOR,
+    // PoseStrategy.LOWEST_AMBIGUITY)
+    //           .get();
+    // var visionPose =
+    //   VisionHelper.update(
+    //           result, tagFieldLayout, PoseStrategy.MULTI_TAG_PNP_ON_COPROCESSOR,
+    // PoseStrategy.LOWEST_AMBIGUITY)
+    //       .get()
+    //       .estimatedPose;
+    // Logger.recordOutput("Vision Pose", visionPose);
+    //   VisionMeasurement measurement = new VisionMeasurement(estPose,
+    // VisionHelper.findVisionMeasurements(estPose));
+    //     while (measurement != null) {
+    //         var estimation = measurement.estimation();
+    //         var estimatedPose = estimation.estimatedPose;
+    //         // Check height of final pose for sanity. Robot should never be more than 0.5 m off
+    // the ground.
+    //         if (Math.abs(estimatedPose.getZ()) > 0.5) {
+    //             continue;
+    //         }
+    //         // Skip single-tag measurements with too-high ambiguity.
+    //         if (estimation.targetsUsed.size() < 2
+    //                 &&
+    // estimation.targetsUsed.get(0).getBestCameraToTarget().getTranslation().getNorm() >
+    // Units.feetToMeters(13)) {
+    //             continue;
+    //         }
+    //         estimator.addVisionMeasurement(
+    //                 measurement.estimation().estimatedPose.toPose2d(),
+    //                 measurement.estimation().timestampSeconds,
+    //                 measurement.confidence());
+    //         Logger.recordOutput("Odo + Vision Pose", estimator.getEstimatedPosition());
+    //       }
+    //     } catch (NoSuchElementException e) {
+    //     }
+    Logger.recordOutput(
```

Comment:
this can be removed bc getPose exists

Reply:
by "this" do you mean line 281

Reply:
yes since the swerve pose is alr logged w `@AutoLogOutput`